service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/PbcService.kt

```diff
@@ -203,4 +211,28 @@ class PbcService(
                 .toAny()
                 .toTxBody()
         ).throwIfFailed("Accept failed")
+
+    private val maxExpiration = OffsetDateTime.of(2031, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC)
+
+    fun grantAuthz(coins: List<Coin>, expiration: OffsetDateTime?) =
+        grpcClientService.new().estimateAndBroadcastTx(
+            signers = listOf(BaseReqSigner(managerKey)),
+            txBody = MsgGrant.newBuilder()
+                .setGranter(managerAddress)
+                .setGrantee(provenanceProperties.contractAddress)
+                .setGrant(
+                    Grant.newBuilder()
+                        .setExpiration((expiration ?: maxExpiration).toProtoTimestamp())
+                        .setAuthorization(
+                            MarkerTransferAuthorization.newBuilder()
+                                .also { builder -> coins.forEach { builder.addTransferLimit(it) } }
```

Comment:
is there a `addAllTransferLimit` function?

Reply:
doh there has to be